timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -102,6 +102,9 @@ def __init__(self, data: Any, static_features: Optional[pd.DataFrame] = None, *a
                 self._validate_multi_index_data_frame(data)
             else:
                 data = self._construct_pandas_frame_from_data_frame(data)
+            # Make sure that timestamps are sorted but item_id order is preserved
+            item_ids = data.index.unique(level=ITEMID)
+            data = data.sort_values(by=TIMESTAMP).loc[item_ids]
```

Comment:
We could add a flag to the `TimeSeriesDataFrame` constructor and set it to `False` when creating the `TimeSeriesDataFrame` internally (e.g., inside `slice_by_timestep`) to avoid redundant sorting every time we slice inside the evaluator (this happens a lot when we're fitting the ensemble).